lib/cfn-model/parser/api_gateway_stage_parser.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require_relative 'parser_error'
+require 'cfn-model/model/api_gateway_stage'
+require 'cfn-model/model/references'
+
+class ApiGatewayStageParser
+
+  def parse(cfn_model:, resource:)
+    api_stage = resource
+
+    attach_usage_plans_to_api_stage(cfn_model: cfn_model, api_stage: api_stage)
+    attach_deployment_id_to_api_stage(cfn_model: cfn_model, api_stage: api_stage)
+    api_stage
+  end
+
+  private
+
+  def attach_usage_plans_to_api_stage(cfn_model:, api_stage:)
+    usage_plans = cfn_model.resources_by_type 'AWS::ApiGateway::UsagePlan'
+    usage_plans.select do |usage_plan|
```

Comment:
these select blocks have a side effect.... if you are just iterating and building up an external data structure, i would use "each" instead.